static/js/src/advantage/manage-subscription.js

```diff
@@ -1,93 +1,259 @@
+    cancelling_subscription_failed:
+      "<strong>Cancelling subscription failed:</strong> Contact <a class='p-notification__action' href='https://ubuntu.com/contact-us'>Canonical sales</a> if the problem persists.",
+    pending_purchase:
+      "<strong>Error:</strong> You already have a pending purchase. Please go to <a href='/advantage/payment-methods'>payment methods</a> to fix it. Reloading page...",
+    unknown_error:
+      "<strong>Unknown error:</strong> Contact <a class='p-notification__action' href='https://ubuntu.com/contact-us'>Canonical sales</a> if the problem persists.",
+  };
 
-function handleAPICall(APIFunction, parameters, button) {
-  const buttonText = button.innerHTML;
-  button.classList.add("is-processing");
-  button.innerHTML =
-    '<i class="p-icon--spinner u-animation--spin is-light"></i>';
-  button.disabled = true;
+  if (map[code]) {
+    return map[code];
+  }
 
-  APIFunction(...parameters)
-    .then((data) => {
-      if (data.errors) {
-        console.error(data.errors);
-        button.classList.remove("is-processing");
-        button.innerHTML = buttonText;
-        button.disabled = false;
-      } else {
-        // if it reloads too fast then some data is missing.
-        setTimeout(function () {
-          location.reload();
-        }, 2000);
+  return default_message;
+};
+
+const handleSuccess = (elements, code) => {
+  const message = getMessage(code, "Success!");
+
+  elements["button"]["html"].classList.remove("is-processing");
+  elements["button"]["html"].innerHTML = elements["button"]["text"];
+  elements["button"]["html"].disabled = false;
+  elements["success"].forEach((element) => {
+    element.classList.remove("u-hide");
+    element.querySelector(".p-notification__message").innerHTML = message;
+  });
+
+  setTimeout(() => {
+    location.reload();
+  }, 2000);
+};
+
+const handleError = (elements, code) => {
+  const message = getMessage(code, "Error!");
+
+  elements["button"]["html"].classList.remove("is-processing");
+  elements["button"]["html"].innerHTML = elements["button"]["text"];
+  elements["button"]["html"].disabled = false;
+  elements["caution"].forEach((element) => {
+    element.classList.remove("u-hide");
+    element.querySelector(".p-notification__message").innerHTML = message;
+  });
+};
+
+const authenticate_3ds = (invoice, elements) => {
+  stripe.confirmCardPayment(invoice.pi_secret).then(function (data) {
+    if (data.error) {
+      handleError(elements, "resizing_machines_fail");
+
+      return;
+    }
+
+    handleSuccess(elements, "resizing_machines_success");
+  });
+};
+
+// It take a time for the invoice to be attached to the payment.
+// We try 5 times each 2 seconds.
+const retryGetPurchase = (purchase_id, attempts_counter, elements) => {
```

Comment:
```suggestion
const retryGetPurchase = (purchaseID, attemptsCounter, elements) => {
```
Can you use camelCase, please? :camel:

Reply:
I did not know before today that JS is into :camel: